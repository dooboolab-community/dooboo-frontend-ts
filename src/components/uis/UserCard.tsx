import React, {ReactElement} from 'react';

import {device} from '../../theme';
import styled from '@emotion/styled';
import {useAppContext} from '../../providers/AppProvider';

const ContentWrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-self: center;
  @media ${device.mobileS} {
    max-width: 768px;
    height: 100vh;
    width: 100vw;
    justify-content: center;
    align-items: center;
  }

  @media ${device.tablet} {
    justify-content: flex-start;
    align-items: flex-start;
    margin-top: 400px;
  }
`;

const Text = styled.span`
  font-size: 18px;
  line-height: 1.5;
  font-family: sans-serif;
  color: ${(props) => props.theme.fontColor};
`;

function UserCard(): ReactElement {
  const {state} = useAppContext();

  return (
    <ContentWrapper>
      <Text>{state.user.displayName}</Text>
      <Text>{state.user.age ? state.user.age : ''}</Text>
      <Text>{state.user.job}</Text>
    </ContentWrapper>
  );
}

export default UserCard;
