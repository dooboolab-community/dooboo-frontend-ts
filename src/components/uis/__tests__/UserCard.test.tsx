import * as renderer from 'react-test-renderer';

import React from 'react';
import UserCard from '../UserCard';
import {createTestElement} from '../../../../test/testUtils';

const props = {};
const component = createTestElement(<UserCard {...props} />);

describe('[Temp] render', () => {
  it('renders without crashing', () => {
    const rendered = renderer.create(component).toJSON();

    expect(rendered).toBeTruthy();
  });
});
