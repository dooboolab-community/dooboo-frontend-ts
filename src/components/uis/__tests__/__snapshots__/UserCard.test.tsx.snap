// Vitest Snapshot v1

exports[`[Temp] render > renders without crashing 1`] = `
<div
  style={
    {
      "flex": 1,
      "textAlign": "center",
      "width": "100vw",
    }
  }
>
  <div
    className="css-1vx0ptc"
  >
    <span
      className="css-1l2m8ze"
    />
    <span
      className="css-1l2m8ze"
    />
    <span
      className="css-1l2m8ze"
    />
  </div>
</div>
`;
